daily-challenges/2024-10/26/ththao27.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { test, expect } from '@playwright/test';
+
+type Product = {
+    name: string;
+    quantity: number;
+};
+
+async function getShoppingList (page: any): Promise<Product[]> {
+    const shoppingList: Product[] = [];
+    const rows = await page.locator('#wishlist > tr').all();
+    for (const row of rows) {
+        const cells = await row.locator('td').allTextContents();
+        shoppingList.push({ name: cells[0], quantity: Number(cells[1]) });
+    }
+    return shoppingList;
+}
+
+async function clickAddBtn (page: any, product: Product) {
+    const plusButton = await page.locator(`#cart div:has-text("${product.name} - +")`).getByRole('button').first();
+    for (let i = 0; i < product.quantity; i++) {
+        await plusButton.click();
```

Comment:
Có thể tối ưu hơn:
- Không dùng vòng for, dùng .click({ clickCount: product.quantity})